portal/server/routes/contract/about/buyerPage.js

```diff
@@ -66,7 +66,7 @@ const buyerSubmissionDetailsProperties = [
 router.post('/contract/:_id/about/buyer', async (req, res) => {
   const { _id, userToken } = requestParams(req);
 
-  const submissionDetailsPayload = constructPayload(req.body, buyerSubmissionDetailsProperties);
+  const submissionDetailsPayload = constructPayload(req.body, buyerSubmissionDetailsProperties,true);
```

Comment:
same q about allowing empty properties here

Reply:
It will insert the empty properties but when we insert the value, the property will update and code will check the validation also for all the pages.

Reply:
ConstructPayload function can expect canPropertyBeEmpty variable. Default the variable is false instead of false we passed true. So constructPayload function will construct the payload variables with empty value. That payload will save in DB object. Second time we come to the same page we fetch all the details from DB for the page and check the value is exist or not. If value is null then will display the error otherwise no error will display. When we enter the value for the empty object, the post will construct new payload and update the DB.

Reply:
I get you. It feels really odd this is needed now. Will close other threads so this ones the only one on the go about it. I'm going to do a bit of testing tomorrow because there's some odd behaviour going on in the existing component on `main` which i feel ties into this